fendermint/actors/bucket/src/actor.rs

```diff
@@ -267,10 +271,84 @@ impl Actor {
             Ok(object.metadata)
         })?;
 
-        emit_evm_event(rt, ObjectMetadataUpdated::new(&params.key, &metadata))?;
+        emit_evm_event(rt, sol::ObjectMetadataUpdated::new(&params.key, &metadata))?;
 
         Ok(())
     }
+
+    fn invoke_contract(
+        rt: &impl Runtime,
+        params: InvokeContractParams,
+    ) -> Result<InvokeContractReturn, ActorError> {
+        let input_data: InputData = params.try_into()?;
+        if sol::can_handle(&input_data) {
+            let output_data = match sol::parse_input(&input_data)? {
+                sol::Calls::addObject_0(call) => {
+                    // function addObject(bytes32 source, string memory key, bytes32 hash, bytes32 recoveryHash, uint64 size) external;
+                    let params = call.params(rt);
+                    Self::add_object(rt, params)?;
+                    call.returns(())
+                }
+                sol::Calls::addObject_1(call) => {
+                    // function addObject(AddObjectParams memory params) external;
+                    let params = call.params(rt);
+                    Self::add_object(rt, params)?;
+                    call.returns(())
+                }
```

Comment:
looks odd to have one method that takes a wrapper params and the other the raw values. imo the params style feels cleaner

Reply:
The wrapper version is slightly less gas-efficient, so it makes sense to have the two IMO.

Reply:
do other methods take a wrapper struct? to your point, if wrapper-less is more efficient, we may as well just not use wrappers anywhere in the sol methods. ie, optimize for gas.

Reply:
And that would be nice! Have it in my branch. Figuring out the fmt and publishing.